src/client/dfuse/ops/mknod.c

```diff
@@ -33,7 +33,6 @@ dfuse_cb_mknod(fuse_req_t req, struct dfuse_inode_entry *parent,
 		D_GOTO(err, rc);
 
 	strncpy(ie->ie_name, name, NAME_MAX);
```

Comment:
so you remove this last character NULL termination because the struct is previously memset to 0?

Reply:
do we need to reserve space for the NULL byte?  e.g. NAME_MAX + 1 ?  At least on my system, NAME_MAX is 255, I assume so the buffer is 256.

Reply:
ah, I guess this is wrong... we only copy NAME_MAX chars at most and it doesn't put NULL character on, so it's correct.

Reply:
yes this is fairly common and the same as in dfs, NAME_MAX is 255, and the array is NAME_MAX+1 in size, so there's a byte on the end which is always zero and the struct is well-aligned etc.  With this in mine strncpy(..., MAX_NAME) is right as it'll always either add a 0 byte itself for short strings of the existing one at the end will be used for max length strings.

What I really don't like however is that the new compiler check is enforcing inconsistent use:

1. When copying a char * to a char[256] it insists you use strncpy(..., 255)
2. When copying a char[256] to a char[256] it insiste you use strncpy(..., 256)

Both of these are safe, and the explicit '\0' in index 255 is only needed for the first case, and only if the buffer isn't initialised to zero previously (so essentially it's a stack variable rather than something in a descriptor).